### PR TITLE
Add new Travis examples

### PR DESCRIPTION
**Pull request by jainvikas8**
### Description

Add the following examples to be built as part of Travis job
- Test kvstore example (K64F)
- Test ble example (NRF52840_DK)
- Test cellular example (WIO_3G)
- Test devicekey example (K66F)
- Test lorawan example (K66F)
- Test crypto example (K64F)
- Test nfc example (DISCO_L475VG_IOT01A)
- Test sockets example (K64F)

### Test Coverage

- [x ]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

**Review by Patater — COMMENTED** [review]
Looks alright. Why WIP?

**Comment by jainvikas8** [comment]
> Looks alright. Why WIP?

I'm seeing a Travis failure.

File: .travis.yml [hunk 1]
```diff
@@ -59,9 +59,17 @@ matrix:
...
+          if [ -z ${SUBEXAMPLE_NAME} ]; then
+             cd ${EXAMPLE_NAME};
+           else
+             cd ${EXAMPLE_NAME}/${SUBEXAMPLE_NAME};
+          fi
         - git clone --depth=1 --single-branch --branch feature-cmake https://github.com/ARMmbed/mbed-os.git
       script:
+        # Replace the content of mbed-os.lib to stay on ${TRAVIS_PULL_REQUEST}
+        - echo “” > mbed-os.lib
+        - mbedtools checkout;
```
[the remark this answers is not in the document]
Comment (jainvikas8) [reply .travis.yml:72]: I'm not sure, as the other command has it below
`mbedtools configure -t GCC_ARM -m ${TARGET_NAME};`

[the remark this answers is not in the document]
Comment (Patater) [reply .travis.yml:72]: I believe it's not needed below either. We shouldn't fix it in this commit though (needs a separate commit). We also shouldn't add new errors.

**Comment by jainvikas8** [comment]
Rebased

**Comment by jainvikas8** [comment]
Refactoring Travis yml

**Comment by jainvikas8** [comment]
Refactor done, ready for review.

File: .travis.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ matrix:
         - cd ${EXAMPLE_NAME}
         - git clone --depth=1 --single-branch --branch feature-cmake https://github.com/ARMmbed/mbed-os.git
       script:
-        - mbedtools configure -t GCC_ARM -m ${TARGET_NAME};
+        - mbedtools configure -t GCC_ARM -m ${TARGET_NAME}
```
Comment (Patater) [inline .travis.yml]: Commit message has a typo:
```diff
- Tarvis yml
+ travis.yml
```

File: travis-ci/functions.sh [hunk 3]
```diff
@@ -69,4 +69,32 @@ _install_gcc_and_ccache()
...
+
+  arm-none-eabi-gcc --version
+  pip install --upgrade cmake
+}
+
+_config_and_build()
+{
+    echo “” > mbed-os.lib
+
+    mbedtools checkout
```
Comment (Patater) [inline travis-ci/functions.sh:81]: Should the following be part of the cloning dependencies function instead?

```
echo “” > mbed-os.lib
mbedtools checkout
```

**Review by Patater — CHANGES_REQUESTED** [review]

File: travis-ci/functions.sh (hunk 3 above)
Comment (jainvikas8) [reply travis-ci/functions.sh:81]: Fixed

**Comment by jainvikas8** [comment]
Forced pushed to fix comments from @Patater

File: .travis.yml (hunk 2 above)
Comment (Patater) [reply .travis.yml]:
```diff
- ci: Remove unnecessary `;` inclusion in .tarvis.yml
+ ci: Remove unnecessary semicolon in .travis.yml
```

**Comment by jainvikas8** [comment]
Forced pushed to fix comments from @Patater

**Review by Patater — APPROVED** [review]

**Pull request merged**